JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -89,6 +89,89 @@ module Microsoft.ApplicationInsights {
         }
 
         /**
+         * Gets the localStorage object if available
+         * @return {Storage} - Returns the storage object if available else returns null
+         */
+        private static _getSessionStorageObject(): Storage {
+            try {
+                if (window.sessionStorage) {
+                    return window.sessionStorage;
+                } else {
+                    return null;
+                }
+            } catch (e) {
+                console.warn('Failed to get client session storage: ' + e.message);
+                return null;
+            }
+        }
+
+        /**
+         *  Check if the browser supports local storage.
+         *
+         *  @returns {boolean} True if local storage is supported.
+         */
+        public static canUseSessionStorage(): boolean {
+            return !!Util._getSessionStorageObject();
+        }
+
```

Comment:
As we discussed - I have mixed feelings about the code duplication here. On one hand it isnt code that is expected to change and this is safest in terms of preventing regressions.  On the other, it would probably be cleaner to add some helper methods and have the same general code path for both local and session storage.

I'm OK with this - but let me know if you have other thoughts now that you are looking at it.